js/ui/grid_core/ui.grid_core.header_filter_core.js

```diff
@@ -360,6 +357,28 @@ export const HeaderFilterView = modules.View.inherit({
         }
     },
 
+    _normalizeHeaderFilterOptions(options) {
+        const generalHeaderFilter = this.option('headerFilter') || {};
+        const specificHeaderFilter = options.headerFilter || {};
+
+        const generalDeprecated = {
+            search: {
+                enabled: generalHeaderFilter.allowSearch,
+                timeout: generalHeaderFilter.searchTimeout,
+            }
+        };
+
+        const columnDeprecated = {
+            search: {
+                mode: specificHeaderFilter.searchMode,
+                enabled: specificHeaderFilter.allowSearch,
+                timeout: specificHeaderFilter.searchTimeout
+            }
+        };
+
+        return extend(true, {}, generalHeaderFilter, generalDeprecated, specificHeaderFilter, columnDeprecated);
+    },
+
```

Comment:
Our option manager already has logic to normalize deprecated options: https://github.com/DevExpress/DevExtreme/blob/bfb0effb8e9d5ffe280bf6d9b224017e366b0167/js/core/options/index.js#L103

Please check that this logic works as we need, and in this case remove `generalDeprecated` and `columnDeprecated`

Reply:
discussed it, it doesn't support array options like column[].headerFilter